flash/vision/detection/model.py

```diff
@@ -57,23 +70,25 @@ class ObjectDetector(Task):
     def __init__(
         self,
         num_classes: int,
-        model: Union[str, nn.Module] = "fasterrcnn_resnet50_fpn",
+        model: str = "fasterrcnn",
+        backbone: Optional[str] = None,
+        fpn: bool = True,
+        pretrained: bool = True,
+        pretrained_backbone: bool = True,
+        trainable_backbone_layers: int = 3,
```

Comment:
This logic `could/should` be within FinetuningCallback.

If the user requires `model= fasterrcnn`, then it should choose the FasterRCNNFinetuning Callback.

Reply:
Both the models will have the same `FineTuningCallback` as they would have similar backbones, but different heads. But yes, could think of moving `trainable_backbone_layers` for `FineTuningCallback` OR we could offer some options for `finetuning` functionalities to the User and it would override the `trainable_backbone_layers`.